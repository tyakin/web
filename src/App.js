import React, { Component } from 'react'

import api from './Api'

import './App.css'

class App extends Component {
      state = {
        user: null,
        email: '',
        password: '',
      }

      loginUser = async (e) => {
        e.preventDefault()

        const { email, password } = this.state

        const { data: token } = await api.loginUser(email, password)

        const { data: user } = await api.getUser(token)

        this.setState({ user })
      }

      inputChange = (e) => this.setState({ [e.target.id]: e.target.value })

      renderLoginForm = () => {
        const { email, password } = this.state

        return (
          <form onSubmit={this.loginUser}>
            <h1>An Awesome Form on React</h1>
            <label htmlFor="email">
              <span>Email:</span>
              <input onChange={this.inputChange} value={email} id="email" name="email" type="email" />
            </label>
            <br />
            <label htmlFor="password">
              <span>Password:</span>
              <input onChange={this.inputChange} value={password} id="password" name="password" type="password" />
            </label>
            <br />
            <button type="submit">Login</button>
          </form>
        )
      }

      render () {
        const { user } = this.state

        return (
          <div className="App">
            {user ? (
              <>
                <div>{`Welcome ${user.email}!`}</div>
                <div>{`Your name is ${user.firstname} ${user.lastname}`}</div>
              </>
            ) : this.renderLoginForm()}
          </div>
        )
      }
}

export default App
