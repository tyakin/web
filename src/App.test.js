import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import App from './App'

configure({ adapter: new Adapter() })

describe('<App />', () => {
  describe('Snapshots', () => {
    it('should render the login form if no user is present', () => {
      const rendered = shallow(<App />)
      expect(rendered.debug()).toMatchSnapshot()
    })
  })
})
