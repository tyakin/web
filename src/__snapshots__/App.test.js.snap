// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Snapshots should render the login form if no user is present 1`] = `
"<div className=\\"App\\">
  <form onSubmit={[Function]}>
    <h1>
      An Awesome Form on React
    </h1>
    <label htmlFor=\\"email\\">
      <span>
        Email:
      </span>
      <input onChange={[Function]} value=\\"\\" id=\\"email\\" name=\\"email\\" type=\\"email\\" />
    </label>
    <br />
    <label htmlFor=\\"password\\">
      <span>
        Password:
      </span>
      <input onChange={[Function]} value=\\"\\" id=\\"password\\" name=\\"password\\" type=\\"password\\" />
    </label>
    <br />
    <button type=\\"submit\\">
      Login
    </button>
  </form>
</div>"
`;
